pysat/_instrument.py

```diff
@@ -534,12 +534,15 @@ def __setitem__(self, key, new):
                 elif len(np.shape(in_data)) == 0:
                     self.data[key] = ('time', [in_data]*len(self.index))
                 else:
-                    # multidimensional input
-                    # user needs to provide what is required
-                    if isinstance(in_data, tuple):
+                    try:
```

Comment:
Is try/except the best way to do this?  What about turning it around:

`if isinstance(in_data, tuple):`
`    self.data[key] = in_data`
`else:`
`    try:`
`        self.data[key] = in_data`
`    except err:`
`        raise ValueError('unknown dimensions for data: {:}'.format(err))`